### PR TITLE
Refactor things method to hit ES directly and accept identifierType param

### PR DESCRIPTION
**Pull request by phamann**
We realised that this method could go directly to ES rather than via the interface. Therefore I've refactored the controller to do just that. We also have a need within myFT to search for a list of things based on their human readable `prefLabel` string instead of the `id` as HUI only returns these and not concept ids :sadpandaface: Therefore have added an `identifierType` param to declare which property you wish the consumer wishes to search by.

#### TL;DR
- Refactor logic to hit ES directly
- Add `identifierType` param as search key
- Refactor tests
- Update readme

/cc @matthew-andrews @ironsidevsquincy

### CODE REVIEW

**Comment by matthew-andrews** [comment]
beautiful work, as ever
